frontend/src/app/settings/index.js

```diff
@@ -0,0 +1,121 @@
+import React, { useRef, useState } from "react"
+import TextField from "@material-ui/core/TextField"
+import Button from "@material-ui/core/Button"
+import { ContentContainer } from "@dom-components/content-container"
+import FormControl from "@material-ui/core/FormControl"
+import { useCommunityService } from "../../services/community"
+import { useCommunityStore } from "@store/community"
+import { LOGIN } from "@store/community/actions"
+import Alert from "@material-ui/lab/Alert"
+import { useHistory } from "react-router-dom"
+import { useSnackbar } from "notistack"
+
+export const SettingsPage = () => {
+	const { changePassword } = useCommunityService()
+	const { enqueueSnackbar } = useSnackbar()
+
+	const [detail, setDetails] = useState("")
+	const [fieldErrors, setFieldErrors] = useState({})
+	const currentPasswordInput = useRef()
+	const newPasswordInput = useRef()
+	const confirmPasswordInput = useRef()
+
+	const clearFields = () => {
+		currentPasswordInput.current.value = ""
+		newPasswordInput.current.value = ""
+		confirmPasswordInput.current.value = ""
+		setFieldErrors({})
+	}
+
+	const saveSettings = (event) => {
+		event.preventDefault()
+		const newPassword = newPasswordInput.current.value
+		const newPasswordConfirm = confirmPasswordInput.current.value
+		clearFields()
+		if (newPassword !== newPasswordConfirm) {
+			setFieldErrors({
+				confirm_password: "Passwords do not match",
+			})
+			return
+		}
+
+		if (!newPassword) {
+			setFieldErrors({
+				new_password: "New password cannot be empty",
+			})
+			return
+		}
+
+		changePassword(currentPasswordInput.current.value, newPassword).then(
```

Comment:
I see the exactly same request pattern accross multiple components. Maybe we should consider abstracting it?

Reply:
Tried it once, it turned into incomprehensible mess.
I might have a better abstraction in mind right now, will revisit this later.